fdbrpc/HTTP.actor.cpp

```diff
@@ -148,11 +225,13 @@ ACTOR Future<size_t> read_delimited_into_string(Reference<IConnection> conn,
 
 	loop {
 		size_t endPos = buf->find(delim, sPos);
-		if (endPos != std::string::npos)
+		if (endPos != std::string::npos) {
 			return endPos - pos;
+		}
 		// Next search will start at the current end of the buffer - delim size + 1
-		if (sPos >= lookBack)
-			sPos -= lookBack;
+		if (buf->size() >= lookBack) {
```

Comment:
this is actually a bug fix i discussed with steve that happens if the http headers get torn across multiple network messages that was found by the buggification in this change